pkg/distributor/distributor.go

```diff
@@ -900,6 +897,64 @@ func extractSampleSeries(
 	return result, bytesRelabelDropped, profilesRelabelDropped
 }
 
+func processServiceNameLabels(labels []*typesv1.LabelPair) []*typesv1.LabelPair {
+	hasServiceName := false
+	hasDotServiceName := false
+	var serviceDotNameIndices []int
+	var serviceDotNameValue string
+	var serviceNameIndex int
+
+	for j, label := range labels {
+		if label.Name == phlaremodel.LabelNameServiceName {
+			hasServiceName = true
+			serviceNameIndex = j
+		} else if label.Name == "service.name" {
+			hasDotServiceName = true
+			serviceDotNameIndices = append(serviceDotNameIndices, j)
+			serviceDotNameValue = label.Value
+		}
+	}
+
+	if hasDotServiceName {
+		if hasServiceName && serviceDotNameValue != "" {
+			labels[serviceNameIndex].Value = serviceDotNameValue
```

Comment:
Do we want this? I've decided `service.name` value to replace `service_name` if both exists.